### PR TITLE
Added the data prepper 1.2.1 performance testing blog post

### PR DESCRIPTION
**Pull request by sbayer55**
Signed-off-by: Steven Bayer <smbayer@amazon.com>

### Description
Added the data prepper 1.2.1 performance testing blog post
 
### Issues Resolved
n/a

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by stockholmux** [comment]
@sbayer55 hey this looks like a pretty good post! I don't see an issue for this though - not sure how to prioritize. Would you mind [putting in a new blog issue](https://github.com/opensearch-project/project-website/issues/new?assignees=&labels=new+blog%2C+enhancement&template=blog_post.md&title=) so we can put it in context with the rest of the blog posts?

**Comment by sbayer55** [comment]
@stockholmux thanks for the feedback. I opened a github issue [#645](https://github.com/opensearch-project/project-website/issues/645)

**Comment by jkowall** [comment]
Skimmed the blog post, is there a reason you didn't compare the throughput with things like vector, fluentd, fluentbit, or OpenTelemetry collector? Comparing against Logstash is somewhat legacy in nature...

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+| Worker Thread        |           12 |       12 |
+| SSL                  |     Disabled | Disabled |
+
+_[*] Note, Max Connection Count, Request Timeout, Buffer Size are not configurable with Logstash 7.13.2_
+
+## Data Prepper Scaling
+
+In this simulation we ramp up from 1 client to 60 over one hour to measure the impact concurrent clients have on performance.
+
+![Data Prepper Throughput Increasing Users](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Graph-Throughput-Increasing-Users.png){: .img-fluid}
```
Comment (dlvenable) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:68]: This image isn't loading when I view the Markdown in your repo.

https://github.com/opensearch-project/project-website/blob/054e4845fa01d6b46adc39fa67e0effa50362ddd/_posts/2022-01-24-data-prepper-1.2.1-performance-testing.md#data-prepper-scaling

**Review by dlvenable — CHANGES_REQUESTED** [review]
Thanks for creating this post!

I made an inline comment about a missing image. Other than that, it looks good to me!

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+| Buffer Size          |    2,000,000 |    n/a * |
+| Batch Size           |        5,000 |    5,000 |
+| Worker Thread        |           12 |       12 |
+| SSL                  |     Disabled | Disabled |
+
+_[*] Note, Max Connection Count, Request Timeout, Buffer Size are not configurable with Logstash 7.13.2_
+
+## Data Prepper Scaling
+
+In this simulation we ramp up from 1 client to 60 over one hour to measure the impact concurrent clients have on performance.
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:66]: ambiguous plural pronouns ('we')

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md (hunk 1 above)
Comment (stockholmux) [reply _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:68]: Looks like the filename has a typo:
filename: Graph-Throughput-**Increating**-Users.png
markdown: Graph-Throughput-**Increasing**-Users.png

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+![Data Prepper Throughput Increasing Users](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Graph-Throughput-Increasing-Users.png){: .img-fluid}
+
+As the number of clients increases Data Preppers throughput remains constant, processing an average of 10,473 MB/m.
+
+# Limitation of Testing
+
+It’s important to note that Data Prepper 1.2.1 and Logstash 7.13.2 support different feature sets, and our performance test suite is targeted at common functionality. As Data Prepper adds more processors and sources, we will expand our to cover those use cases, and we will continue update the community with performance benchmarks.
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:74]: - ambiguous plural pronouns ('our', 'we')
- You will probably change this when rephrasing the pronouns but this doesn't make sense: 'we will expand our to cover those use cases' (ours?)

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+## Candidates for future performance testing scenarios and improvements
+
+![Environment Reference](/assets/media/blog-images/2022-01-25-data-prepper-1.2.1-performance-testing/Environment-Reference.png){: .img-fluid}
+
+In a real-world deployment, if Data Prepper is unable to keep pace with the logs generated by the source application. Data Prepper will become a bottleneck causing backpressure on the source application. In the future we will enhance our performance tests to simulate backpressure and measure the impact.
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:80]: - ambiguous plural pronouns ('we','our')

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+The scope of our initial performance testing scenarios was focused on a common scenario, http source, grok processor, and an OpenSearch sink. As new features are added to Data Prepper in upcoming releases, we plan to add matching performance testing simulations for core functionality.
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:82]: - ambiguous plural pronouns ('we','our')
- Should http source and grok processor be linked to something? Will it be immediately understood by the reader what these are?

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+# Running Data Prepper Performance tests
+Performance suite source code is available on [GitHub](https://github.com/sbayer55/gatling-tests/). To run the full test suite execute `./gradlew –rerun-tasks gatlingRun -Dhost=<target url>`. After all tests have completed HTML reports will be created in `./build/reports/gatling/<simulation-name>-<unix-timestamp>/index.html`. Further instructions on running performance tests and Gatling are available in the repository [readme](https://github.com/sbayer55/gatling-tests/blob/main/README.md).
+
+For more information on Gatling checkout their [website](https://gatling.io/).
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:87]: I would drop this line and just link Gatling in the previous sentence.

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+Table [1] - AWS Environment Details
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:89]: Add a line break after this - it's breaking the site markdown processor
Screen Shot 2022-01-25 at 11 13 42 AM

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+The scope of our initial performance testing scenarios was focused on a common scenario, http source, grok processor, and an OpenSearch sink. As new features are added to Data Prepper in upcoming releases, we plan to add matching performance testing simulations for core functionality.
+
+# Running Data Prepper Performance tests
+Performance suite source code is available on [GitHub](https://github.com/sbayer55/gatling-tests/). To run the full test suite execute `./gradlew –rerun-tasks gatlingRun -Dhost=<target url>`. After all tests have completed HTML reports will be created in `./build/reports/gatling/<simulation-name>-<unix-timestamp>/index.html`. Further instructions on running performance tests and Gatling are available in the repository [readme](https://github.com/sbayer55/gatling-tests/blob/main/README.md).
+
+For more information on Gatling checkout their [website](https://gatling.io/).
+
```
Comment (stockholmux) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:88]: Typically there is a conclusion with a call-to-action at the end of the blog post. 

A good structure is 2-4 sentence that answer the following questions:
- What did story did this blog post tell? (Data prepper preformed better than Logstash?)
- Where can a reader out more?
- What should a reader do if they want to get involved?

**Review by stockholmux — CHANGES_REQUESTED** [review]
Nice blog post! 

- A few grammar/style issues that need to be fixed (same issue repeated mostly)
- A couple of technical/markdown issues
- Needs a conclusion.

**Comment by stockholmux** [comment]
@jkowall - I don't want to speak for @sbayer55, but I'm guessing the Logstash comparison is made because in this context it's a drop-in replacement.

Although a head-to-head with the other packages that fill this niche would be very interesting 🤔

File: _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+In a real-world deployment, if Data Prepper is unable to keep pace with the logs generated by the source application. Data Prepper will become a bottleneck causing backpressure on the source application. In the future we will enhance our performance tests to simulate backpressure and measure the impact.
+
+The scope of our initial performance testing scenarios was focused on a common scenario, http source, grok processor, and an OpenSearch sink. As new features are added to Data Prepper in upcoming releases, we plan to add matching performance testing simulations for core functionality.
+
+# Running Data Prepper Performance tests
+Performance suite source code is available on [GitHub](https://github.com/sbayer55/gatling-tests/). To run the full test suite execute `./gradlew –rerun-tasks gatlingRun -Dhost=<target url>`. After all tests have completed HTML reports will be created in `./build/reports/gatling/<simulation-name>-<unix-timestamp>/index.html`. Further instructions on running performance tests and Gatling are available in the repository [readme](https://github.com/sbayer55/gatling-tests/blob/main/README.md).
```
Comment (dinujoh) [inline _posts/2022-01-24-data-prepper-1.2.1-performance-testing.md:85]: We should consider moving the performance suite source code to data prepper repository or some repository under Opensearch for long term maintainability and contribution.

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 10]
```diff
@@ -0,0 +1,102 @@
+---
+layout: post
+title:  "Data Prepper 1.2.1 Performance Testing"
+authors:
+  - sbayer
+date: 2022-01-28
+categories:
+  - technical-post
+
+excerpt: "Following the launch of log pipeline in Data Prepper 1.2, the Data Prepper Team is excited to share the results of our performance testing suite. Our goal is to create a tool that can simulate a set of real-world scenarios in any environment while maintaining compatibility with popular log ingestion applications."
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:10]: "our" = [ambiguous plural pronoun](https://gist.github.com/stockholmux/52b9b224ad806acd0dd389fe3e79daa0)

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+
+---
+
+Following the launch of log pipeline in [Data Prepper 1.2](https://www.opensearch.org/blog/technical-post/2021/12/Introducing-Data-Prepper-1.2.0-with-Log-Pipelines/), we are excited to share the results of our performance testing suite. Our goal is to create a tool that can simulate a set of real-world scenarios in any environment while maintaining compatibility with popular log ingestion applications. In all performance test results discussed below, the test environments and configurations are identical, except where the same option is not available for all applications. See testing limitations section for additional details.
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:14]: "our" = ambiguous plural pronoun

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+
+In this simulation clients are ramped up from 1 to 60 over one hour to measure the impact concurrent clients have on performance.
+
+![Data Prepper Throughput Increasing Users](/assets/media/blog-images/2022-01-28-data-prepper-1.2.1-performance-testing/Graph-Throughput-Increasing-Users.png){: .img-fluid}
+
+As the number of clients increases Data Preppers throughput remains constant, processing an average of 10,473 MB/m.
+
+# Limitation of Testing
+
+It’s important to note that Data Prepper 1.2.1 and Logstash 7.13.2 support different feature sets, and our performance test suite is targeted at common functionality. As Data Prepper adds more processors and sources, test cases will be added. The Data Prepper team will continue update the community with performance benchmarks.
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:74]: "our" = ambiguous plural pronoun

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+
+## Candidates for future performance testing scenarios and improvements
+
+![Environment Reference](/assets/media/blog-images/2022-01-28-data-prepper-1.2.1-performance-testing/Environment-Reference.png){: .img-fluid}
+
+In a real-world deployment, if Data Prepper is unable to keep pace with the logs generated by the source application. Data Prepper will become a bottleneck causing backpressure on the source application. In the future performance tests will be enhanced to simulate backpressure and measure the impact.
+
+The scope of our initial performance testing scenarios was focused on a common scenario, [http source](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/http-source), [grok processor](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/grok-prepper), and an [OpenSearch sink](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/opensearch). As new features are added to Data Prepper in upcoming releases, performance testing simulations will be added to cover core functionality.
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:82]: "our" = ambiguous plural pronoun

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md (hunk 11 above)
Comment (stockholmux) [reply _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:14]: "we" = ambiguous plural pronoun

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+
+## Data Prepper Environment
+
+![Data Prepper Environment](/assets/media/blog-images/2022-01-28-data-prepper-1.2.1-performance-testing/Data-Prepper.png){: .img-fluid}
+
+## Logstash Environment
+
+![Logstash Environment](/assets/media/blog-images/2022-01-28-data-prepper-1.2.1-performance-testing/Logstash.png){: .img-fluid}
+
+Comparing the performance of the latest release of Data Prepper 1.2.1 against Logstash 7.13.2. We configured 10 clients to send requests as frequently as possible. Each request was 19.2 KB and contained a batch of 200 logs. We then ran this test for 30 minutes and measured the latency and throughput.
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:48]: "we" = ambiguous plural pronoun

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+
+The scope of our initial performance testing scenarios was focused on a common scenario, [http source](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/http-source), [grok processor](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/grok-prepper), and an [OpenSearch sink](https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/opensearch). As new features are added to Data Prepper in upcoming releases, performance testing simulations will be added to cover core functionality.
+
+# Running Data Prepper Performance tests
+
+Performance suite source code is available on [GitHub](https://github.com/sbayer55/gatling-tests/). To run the full test suite execute `./gradlew –rerun-tasks gatlingRun -Dhost=<target url>`. After all tests have completed HTML reports will be created in `./build/reports/gatling/<simulation-name>-<unix-timestamp>/index.html`. Further instructions on running performance tests and [Gatling](https://gatling.io/) are available in the repository [readme](https://github.com/sbayer55/gatling-tests/blob/main/README.md).
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:86]: `https://github.com/sbayer55/gatling-tests/` should not be on a personal github account.

File: _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+categories:
+  - technical-post
+
+excerpt: "Following the launch of log pipeline in Data Prepper 1.2, the Data Prepper Team is excited to share the results of our performance testing suite. Our goal is to create a tool that can simulate a set of real-world scenarios in any environment while maintaining compatibility with popular log ingestion applications."
+
+---
+
+Following the launch of log pipeline in [Data Prepper 1.2](https://www.opensearch.org/blog/technical-post/2021/12/Introducing-Data-Prepper-1.2.0-with-Log-Pipelines/), we are excited to share the results of our performance testing suite. Our goal is to create a tool that can simulate a set of real-world scenarios in any environment while maintaining compatibility with popular log ingestion applications. In all performance test results discussed below, the test environments and configurations are identical, except where the same option is not available for all applications. See testing limitations section for additional details.
+
+# TL;DR Data Prepper 1.2.1 vs Logstash 7.13.2 Results
```
Comment (stockholmux) [inline _posts/2022-01-28-data-prepper-1.2.1-performance-testing.md:16]: I didn't catch this the first time but did you explain why Logstash 7.13.x was used not the current version?

**Review by stockholmux — CHANGES_REQUESTED** [review]
- OpenSearch voice issue
- Github Link
- 7.13.x question

**Review by stockholmux — APPROVED** [review]
LGTM

**Pull request merged**